nuxeo-common/src/main/java/org/nuxeo/common/utils/ZipUtils.java

```diff
@@ -474,4 +474,19 @@ public static boolean hasEntry(URL url, String entryName) throws IOException {
         return hasEntry(url.openStream(), entryName);
     }
 
+    /**
+     * Check if the content of the stream is a valid zip.
+     * @param stream the stream to be validated
+     * @return true if the stream is a valid zip, false otherwise
+     */
+    public static boolean isValid(InputStream stream) {
+        try {
+            ZipInputStream zip = new ZipInputStream(stream);
```

Comment:
Please close the input stream after having used it (maybe using the try-with-resources statement)

Reply:
Did not go through that step as I was not sure if the stream that is used as input would be closed too. Do you have any consideration on this?

Reply:
Hmm, you're right :+1: Let's keep it as it is. Indeed, closing the resources should be caller's responsibility.

Reply:
Ok, makes sense to me. Thanks 👍

Reply:
This should be documented in the method signature ("The stream is not closed"). Also your caller in `IWork2PDFConverter` must then close the stream.

Reply:
On second thought, I'd rather we not add trivial things to nuxeo-common. Please just inline whatever code you need in your caller method and leave nuxeo-common as it is. We should try to *decrease* the size of nuxeo-common, which is often very redundant with other common libraries.

Reply:
Do you mean the caller in `IWork2PDFConverter` should close the `ZipInputStream`? 
Also, I understand the logic of leaving nuxeo-common as it is, but I thought it would be better to encapsulate the access to zip methods in nuxeo-common, because the converter was already using the class `ZipUtils` from nuxeo-common to handle the zip files.

Reply:
Hm yes I didn't have the whole file context. If `ZipUtils` is already used why not.

A method should close whatever stream it creates/acquires. So `IWork2PDFConverter` must close the stream it opens with `blob.getStream()`, and `isValid` should close the `ZipInputStream` it creates.

Reply:
👍  thanks, doing some changes to this then.